tests/phpunit/tests/blocks/renderCommentTemplate.php

```diff
@@ -512,4 +512,97 @@ public function test_rendering_comment_template_unmoderated_preview() {
+		$comment_author_name_block        = new WP_Block(
+			$parsed_comment_author_name_block,
+			array(
+				'commentId' => self::$comment_ids[0],
+			)
+		);
+		$comment_author_name_block_markup = $comment_author_name_block->render();
+
+		$render_block_callback = static function( $block_content, $block ) use ( $parsed_comment_author_name_block ) {
+			// Insert a Comment Author Name block (which requires `commentId`
+			// block context to work) after the Comment Content block.
+			if ( 'core/comment-content' !== $block['blockName'] ) {
+				return $block_content;
+			}
+
+			$inserted_content = render_block( $parsed_comment_author_name_block );
+			return $inserted_content . $block_content;
+		};
+
+		add_filter( 'render_block', $render_block_callback, 10, 3 );
+		$parsed_blocks = parse_blocks(
+			'<!-- wp:comment-template --><!-- wp:comment-content /--><!-- /wp:comment-template -->'
+		);
+		$block         = new WP_Block(
+			$parsed_blocks[0],
+			array(
+				'postId' => self::$custom_post->ID,
+			)
+		);
+		$markup        = $block->render();
+		remove_filter( 'render_block', $render_block_callback );
+
+		$this->assertStringContainsString( $comment_author_name_block_markup, $markup );
+	}
+
+	public function test_inner_block_inserted_by_render_block_data_is_retained() {
+		$render_block_callback = new MockAction();
+		add_filter( 'render_block', array( $render_block_callback, 'filter' ), 10, 3 );
+
+		$render_block_data_callback = static function( $parsed_block ) {
+			// Add a Social Links block to a Comment Template block's inner blocks.
+			if ( 'core/comment-template' === $parsed_block['blockName'] ) {
+				$inserted_block_markup = <<<END
+<!-- wp:social-links -->
+<ul class="wp-block-social-links"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /--></ul>
+<!-- /wp:social-links -->'
+END;
+
+				$inserted_blocks = parse_blocks( $inserted_block_markup );
+
+				$parsed_block['innerBlocks'][] = $inserted_blocks[0];
+			}
+			return $parsed_block;
+		};
+
+		// Remove auto-insertion filter so it won't collide.
+		remove_filter( 'render_block_data', 'gutenberg_auto_insert_child_block' );
+
+		add_filter( 'render_block_data', $render_block_data_callback, 10, 1 );
+		$parsed_blocks = parse_blocks(
+			'<!-- wp:comments --><!-- wp:comment-template --><!-- wp:comment-content /--><!-- /wp:comment-template --><!-- /wp:comments -->'
+		);
+		$block         = new WP_Block(
+			$parsed_blocks[0],
+			array(
+				'postId' => self::$custom_post->ID,
+			)
+		);
+		$block->render();
+		remove_filter( 'render_block_data', $render_block_data_callback );
+		// Add back auto-insertion filter.
+		add_filter( 'render_block_data', 'gutenberg_auto_insert_child_block', 10, 1 );
+
+		$this->assertSame( 5, $render_block_callback->get_call_count() );
+
+		$args = $render_block_callback->get_args();
+		$this->assertSame( 'core/comment-content', $args[0][2]->name );
+		$this->assertSame( 'core/comment-template', $args[1][2]->name );
```

Comment:
These three assertions will need messages to help differentiate between failures.

Reply:
a4add40810